apps/gdalmdimtranslate_lib.cpp

```diff
@@ -1188,29 +1192,65 @@ static bool TranslateArray(
+                        auto var = std::dynamic_pointer_cast<VRTMDArray>(
+                            poDstGroup->CreateMDArray(
+                                newDimName, {dstDim},
+                                GDALExtendedDataType::Create(GDT_Float64)));
+                        if (var)
+                        {
+                            const double dfStart =
+                                srcIndexVar->GetName() == "X"
+                                    ? gt[0] + (range.m_nStartIdx + 0.5) * gt[1]
+                                    : gt[3] + (range.m_nStartIdx + 0.5) * gt[5];
+                            const double dfIncr =
+                                (srcIndexVar->GetName() == "X" ? gt[1]
+                                                               : gt[5]) *
+                                range.m_nIncr;
+                            std::unique_ptr<VRTMDArraySourceRegularlySpaced>
+                                poSource(new VRTMDArraySourceRegularlySpaced(
+                                    dfStart, dfIncr));
+                            var->AddSource(std::move(poSource));
+                            bIndexingVarCreated = true;
+                        }
                     }
                 }
+                if (!bIndexingVarCreated &&
+                    srcIndexVar->GetDimensionCount() == 1 &&
+                    srcIndexVar->GetDataType().GetClass() != GEDTC_COMPOUND &&
+                    srcIndexVar->GetDimensions()[0]->GetSize() < 10000)
+                {
+                    auto var = std::dynamic_pointer_cast<VRTMDArray>(
+                        poDstGroup->CreateMDArray(newDimName, {dstDim},
+                                                  srcIndexVar->GetDataType()));
+
+                    std::vector<GUInt64> anOffset = {0};
+                    const size_t nCount = static_cast<size_t>(
+                        srcIndexVar->GetDimensions()[0]->GetSize());
+                    std::vector<size_t> anCount = {nCount};
+                    std::vector<GByte> abyValues;
+                    abyValues.resize(srcIndexVar->GetDataType().GetSize() *
+                                     nCount);
+                    const GInt64 arrayStep[] = {1};
+                    const GPtrDiff_t anBufferStride[] = {1};
+                    srcIndexVar->Read(anOffset.data(), anCount.data(),
+                                      arrayStep, anBufferStride,
+                                      srcIndexVar->GetDataType(),
+                                      abyValues.data());
+                    auto poSource =
+                        std::make_unique<VRTMDArraySourceInlinedValues>(
+                            var.get(),
+                            /* bIsConstantValue = */ false, std::move(anOffset),
+                            std::move(anCount), std::move(abyValues));
+                    var->AddSource(std::move(poSource));
+                }
```

Comment:
Should we add an `else` and log an error?

Reply:
added a CPLDebug() message if the conditions are not met (which would mean a rather weird source indexing variable. Unlikely to happen in practice)